haas/ext/obm/ipmi_1_5.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2015-2016 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""IPMI driver for implementing out of band management. """
+
+from sqlalchemy import Column, String, Integer, ForeignKey
+import schema
+import subprocess
+
+from haas.model import Obm
+from haas.errors import OBMError
+from haas.dev_support import no_dry_run
+from subprocess import call, check_call, Popen, PIPE
+import os
+
+class Ipmi_1_5(Obm):
+    id = Column(Integer, ForeignKey('obm.id'), primary_key=True)
+    host = Column(String, nullable=False)
+    user = Column(String, nullable=False)
+    password = Column(String, nullable=False)
+
+    api_name = 'http://schema.massopencloud.org/haas/v0/obm/ipmi_1_5'
+
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+        }
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'type': Ipmi_1_5.api_name,
+            'host': basestring,
+            'user': basestring,
+            'password': basestring,
+            }).validate(kwargs)
+
+    def _ipmitool(self, args):
+        """Invoke ipmitool with the right host/pass etc. for this node.
+
+        `args`- A list of any additional arguments to pass to ipmitool.
+        Returns the exit status of ipmitool.
+        """
+        status = call(['ipmitool',
+            '-U', self.user,
```

Comment:
Comment out the user part "-U ..." 
Sun servers at least does not need the user.